surf/apps/communities/models.py

```diff
@@ -89,6 +65,44 @@ def clean(self):
                 raise ValidationError("SURFconext group id isn't a valid URN. Check "
                                       "https://en.wikipedia.org/wiki/Uniform_Resource_Name for examples of valid URNs.")
 
+    # by overriding the save method we can force a clean on the community on an update.
+    def save(self, *args, **kwargs):
+        self.full_clean()
```

Comment:
This is tricky business. There are quite a few ways in which a model will "update" without going through the save method. For instance the new: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#bulk-update, but also the more traditional: https://docs.djangoproject.com/en/3.0/ref/models/querysets/#django.db.models.query.QuerySet.update

Your desire to have a safe, save method means that somewhere in your code you're dealing with unsafe models and you now expect people to know/remember that it's not safe to use any of the update queryset methods and that they have to use "save" instead or do their own full_clean. Instead you could in your code call full_clean and then pass the models on to where ever they need to go knowing that they are safe. That way people don't have to worry about the cleanness of the models.

So in a way, with doing it like this, you create a global state which is: "are the community models clean". You don't create a variable for it, but all the bugs that might creep out of this will be as if you had a community_models_are_clean global variable. Things will go wrong in very unexpected ways. 

So reconsider your approach and clean models immediately after creation (or whatever you do) and don't wait for "save".